doc/modules/grid_search.rst

```diff
@@ -1,26 +1,30 @@
-.. currentmodule:: sklearn.grid_search
+
+.. currentmodule:: sklearn.model_selection
 
 .. _grid_search:
 
-===============================================
-Grid Search: Searching for estimator parameters
-===============================================
+===========================================
+Tuning the hyper-parameters of an estimator
+===========================================
+
+Hyper-parameters are parameters that are not directly learnt within estimators.
+In scikit-learn, they are passed as arguments to the constructor of the estimator
+classes. Typical examples include ``C``, ``kernel`` and ``gamma`` for Support
+Vector Classifier, ``alpha`` for Lasso, etc.
 
-Parameters that are not directly learnt within estimators can be set by
-searching a parameter space for the best :ref:`cross_validation` score.
-Typical examples include ``C``, ``kernel`` and ``gamma`` for Support Vector
-Classifier, ``alpha`` for Lasso, etc.
+They are often referred to as *hyperparameters* (particularly in Bayesian
```

Comment:
Maybe rephrase as:

Hyper-parameters should not be confused with the internal model parameters that are directly learned by the inner machine learning procedure from the data when calling the `fit` or `partial_fit` methods.